DS_and_Algorithm/Selenium/test.java

```diff
@@ -0,0 +1,706 @@
+  }
+
+  @Test
+  void testShouldBeAbleToToggleAnEnabledSelectedCheckbox() {
+    driver.get(pages.formPage);
+    assertCanToggle(enabledSelectedCheckbox());
+  }
+
+  @Test
+  void testClickingOnASelectedRadioButtonShouldLeaveItSelected() {
+    driver.get(pages.formPage);
+
+    WebElement button = enabledSelectedRadioButton();
+    assertThat(button.isSelected()).isTrue();
+
+    button.click();
+
+    assertThat(button.isSelected()).isTrue();
+  }
+
+  @Test
+  void testShouldBeAbleToToggleEnabledMultiSelectOption() {
+    driver.get(pages.formPage);
+    assertCanToggle(selectedMultipleSelectOption());
+  }
+
+  @Test
+  void testShouldBeAbleToToggleSelectableCheckboxByClickingOnIt() {
+    driver.get(pages.formPage);
+
+    WebElement checkbox = enabledUnselectedCheckbox();
+    assertNotSelected(checkbox);
+
+    checkbox.click();
+    assertSelected(checkbox);
+
+    checkbox.click();
+    assertNotSelected(checkbox);
+  }
+
+  @Test
+  void testShouldBeAbleToSelectSelectableRadioButtonByClickingOnIt() {
+    driver.get(pages.formPage);
+
+    WebElement radioButton = enabledUnselectedRadioButton();
+    assertNotSelected(radioButton);
+
+    radioButton.click();
+    assertSelected(radioButton);
+
+    radioButton.click();
+    assertSelected(radioButton);
+  }
+
+  @Test
+  void testClickingDisabledSelectedCheckboxShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(randomlyDisabledSelectedCheckbox());
+  }
+
+  @Test
+  void testClickingDisabledUnselectedCheckboxShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(disabledUnselectedCheckbox());
+  }
+
+  @Test
+  void testClickingDisabledSelectedRadioButtonShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(disabledSelectedRadioButton());
+  }
+
+  @Test
+  void testClickingDisabledUnselectedRadioButtonShouldBeNoop() {
+    driver.get(pages.formPage);
+    assertClickingPreservesCurrentlySelectedStatus(disabledUnselectedRadioButton());
+  }
+
+  private void assertNotSelected(WebElement element) {
+    assertSelected(element, UNSELECTED);
+  }
+
+  private void assertSelected(WebElement element) {
+    assertSelected(element, SELECTED);
+  }
+
+  private void assertSelected(WebElement element, boolean isSelected) {
+    wait.until(ExpectedConditions.elementSelectionStateToBe(element, isSelected));
+    assertThat(element.isSelected())
+        .describedAs(
+            "Expected element %s to be %s",
+            describe(element), selectedToString(isSelected), selectedToString(!isSelected))
+        .isEqualTo(isSelected);
+  }
+
+  private void assertCannotSelect(WebElement element) {
+    boolean previous = element.isSelected();
+    element.click();
+    assertThat(element.isSelected()).isEqualTo(previous);
+  }
+
+  private void assertCanSelect(WebElement element) {
+    assertNotSelected(element);
+
+    element.click();
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/17#issuecomment-2188806524>#1bfcae</a><span> - 06/25/2024, 12:32 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Clicking on an element without checking if it is enabled may lead to unexpected behavior or exceptions. The code does not check if the element is enabled before attempting to click it.
</div>
<div id="fix">
<b>Fix:</b> Add a check to ensure the element is enabled before attempting to click it.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -406,7 +406,9 @@
     assertNotSelected(element);
 +   if (element.isEnabled()) {
     element.click();
 +   }
     assertSelected(element);
```
</div>
</details>
</div>
<br>


</div>